components/eventing-controller/controllers/subscription/nats/unit_test.go

```diff
@@ -0,0 +1,309 @@
+package nats
+
+import (
+	"context"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	kymalogger "github.com/kyma-project/kyma/common/logging/logger"
+	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+	"github.com/kyma-project/kyma/components/eventing-controller/logger"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/mocks"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestHandleSubscriptionDeletion(t *testing.T) {
+	g := NewGomegaWithT(t)
+	testEnvironment := setupTestEnvironment(t)
+	ctx, r, mockedBackend := testEnvironment.Context, testEnvironment.Reconciler, testEnvironment.Backend
+
+	subscription := &eventingv1alpha1.Subscription{
+		ObjectMeta: v1.ObjectMeta{
+			Name:       "test",
+			Namespace:  namespaceName,
+			Finalizers: []string{}, // empty finalizers
+		},
+	}
+
+	err := testEnvironment.Reconciler.Client.Create(testEnvironment.Context, subscription)
+	g.Expect(err).Should(BeNil())
+
+	err = r.handleSubscriptionDeletion(ctx, subscription, r.namedLogger())
+	g.Expect(err).Should(BeNil())
+
+	// the subscription has no kyma eventing finalizers and hence should not be processed in the function
+	mockedBackend.AssertNotCalled(t, "DeleteSubscription", subscription)
+	g.Expect(subscription.ObjectMeta.Finalizers).Should(BeEmpty())
+	g.Expect(err).Should(BeNil())
+
+	// add the eventing finalizer to test the function's branches
```

Comment:
Aren't we testing two different cases here ? 
1. subscription without finalizer and nothing should happen
2. subscription with finalizer and something should happen

If you agree with me we could use a table-driven test here instead.

Reply:
Done